open-api/rest-catalog-open-api.yaml

```diff
@@ -28,6 +28,10 @@ info:
   description:
     Defines the specification for the first version of the REST Catalog API.
     Implementations should ideally support both Iceberg table specs v1 and v2, with priority given to v2.
+
+    This API uses capability tags to describe optional functionality.
+    In order to support a particular capability, a server is required to implement all endpoints grouped under a particular tag.
```

Comment:
I think this is the sentence is what I see a problem, where a server implementing `tables` might only implement read APIs. We can maybe discuss if we should separate those into `table-read`, `table-management`, `table-registration`, etc. and do similar to views. 

But even for that, for example in `table-management`, `RenameTable` might be something that is difficult to do atomically and servers choose to not do that. So "required" still seems wrong. When you say "required to implement", do you consider "always throw 403 or 501" a way to implement the API? If the answer is yes, let's make it clear here. If we divide features into reasonably small chunks, and require the server to at least fail the operation properly when they declare they have the capability, it seems like a good balance to me.

Reply:
If it helps, then I can change this from `required` to `should implement` and servers only partially implementing a capability can highlight this through docs or other meaningful ways, but rather than saying `should implement` the server should be explicit and throw a 501 (not implemented) in that case. I'll update the wording here to make this clearer.

As I mentioned in https://lists.apache.org/thread/qq9bfqtx9yoqb3goj3zvjsgwt8hotvj9 capabilities should rather be coarse-grained and an easy way for a client to understand what capabilities a server support. 
Introducing `table-read` / `table-management` / `table-write` / `table-delete` seems to be rather going in a finer grained direction and I don't think we want to end up having a 1:1 mapping between a single endpoint and a single capability.

Reply:
> If it helps, then I can change this from required to should implement and servers only partially implementing a capability can highlight this through docs or other meaningful ways, but rather than saying should implement the server should be explicit and throw a 501 (not implemented) in that case. I'll update the wording here to make this clearer.

Yes I think that is better!

Reply:
> Introducing table-read / table-management / table-write / table-delete seems to be rather going in a finer grained direction and I don't think we want to end up having a 1:1 mapping between a single endpoint and a single capability.

We discussed this in the last sync meeting. I am good with having just a tables capability + documentation. I mainly raised it after seeing ones like `table-register` recommended in the PR, but seems like that is no longer valid.